streams/src/main/java/org/apache/kafka/streams/processor/assignment/AssignmentConfigs.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.assignment;
+
+import java.util.List;
+
+/**
+ * Configuration for the KafkaStreams Task Assignor.
+ */
+public class AssignmentConfigs {
+
+    private long acceptableRecoveryLag;
+    private int maxWarmupReplicas;
+    private int nonOverlapCost;
+    private int numStandbyReplicas;
+    private long probingRebalanceIntervalMs;
+    private List<String> rackAwareAssignmentTags;
+    private int trafficCost;
+
+    /**
+     * Returns the acceptable lag in the task recovery process in milliseconds.
```

Comment:
I'd just link to the relevant StreamsConfig, where applicable, instead of trying to rewrite the docs for these:

```suggestion
     * The configured acceptable recovery lag according to 
     * {@link StreamsConfig#ACCEPTABLE_RECOVERY_LAG_CONFIG}
```